### ISSUE TITLE
[BUG] Rule is not triggered on setProperty method

### ISSUE THREAD

**Issue by ibragins** [konveyor/analyzer-lsp#854]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Konveyor version

latest

### Priority

Undefined (Default)

### Current Behavior

```
  when:
    java.referenced:
      location: METHOD_CALL
      pattern: java.util.Properties.setProperty(java.lang.String, java.lang.String)
```
Should trigger an incident in the following code (extracted from [tackle-testapp](https://github.com/konveyor/tackle-testapp-public/blob/main/src/main/java/io/konveyor/demo/ordermanagement/config/PersistenceConfig.java#L68))
```
        hibernateProperties.setProperty("hibernate.hbm2ddl.auto", config.getProperty("hibernate.hbm2ddl.auto"));
        hibernateProperties.setProperty("hibernate.dialect", config.getProperty("hibernate.dialect"));
        hibernateProperties.setProperty("hibernate.cache.use_second_level_cache", "false");
```
But it doesn't trigger any incidents.

### How Reproducible

Always (Default)

### Environment

```markdown
- **Analyzer Image SHA** 
- **JDTLS  base image SHA**
- **Provider Settings JSON**
```

**Comment by tsanders-rh** [comment]
## Root Cause Analysis

I've investigated this issue and identified the root cause. The problem is in the **java-analyzer-bundle** repository, specifically in the `CustomASTVisitor.java` class.

### The Bug

When a METHOD_CALL pattern includes parameter types (e.g., `java.util.Properties.setProperty(java.lang.String, java.lang.String)`), the `CustomASTVisitor` fails to match because:

1. The AST resolves method names as `ClassName.methodName` **without** parameter types
2. The query pattern still contains the parameter types `(java.lang.String, java.lang.String)`
3. The regex match fails due to this mismatch

### The Fix

I've created a PR to fix this issue: https://github.com/konveyor/java-analyzer-bundle/pull/179

The fix preprocesses the query pattern to strip parameter types before matching, allowing patterns like:
- `java.util.Properties.setProperty(java.lang.String, java.lang.String)`

To correctly match method calls like:
- `hibernateProperties.setProperty("key", "value")`

### Testing

Once the fix is merged and a new version of java-analyzer-bundle is released, this issue will be resolved. The test case I created confirms that the pattern should trigger on the `setProperty` calls in PersistenceConfig.java.
